File: examples/ios/MobileTrackerExample/MobileTrackerExample/UserDataManager.swift
import Foundation

/// Manages persistence of user data (userId, name, email) across app sessions
/// 
/// Stores user information in UserDefaults so it can be restored when the app reopens,
/// unless the user explicitly resets the session or all data.
struct UserDataManager {
    private static let prefsName = "user_data_prefs"
    private static let keyUserId = "user_id"
    private static let keyUserName = "user_name"
    private static let keyUserEmail = "user_email"
    
    /// Save user data to UserDefaults
    static func saveUserData(userId: String, name: String, email: String) {
        let defaults = UserDefaults.standard
        defaults.set(userId, forKey: keyUserId)
        defaults.set(name, forKey: keyUserName)
        defaults.set(email, forKey: keyUserEmail)
    }
    
    /// Load user data from UserDefaults
    static func loadUserData() -> UserData? {
        let defaults = UserDefaults.standard
        
        guard let userId = defaults.string(forKey: keyUserId) else {
            return nil
        }
        
        let name = defaults.string(forKey: keyUserName) ?? ""
        let email = defaults.string(forKey: keyUserEmail) ?? ""
        
        return UserData(userId: userId, name: name, email: email)
    }
    
    /// Clear user data from UserDefaults
    static func clearUserData() {
        let defaults = UserDefaults.standard
        defaults.removeObject(forKey: keyUserId)
        defaults.removeObject(forKey: keyUserName)
        defaults.removeObject(forKey: keyUserEmail)
    }
}

/// Data structure for user information
struct UserData {
    let userId: String
    let name: String
    let email: String
}

File: examples/android/src/main/java/ai/founderos/mobiletracker/example/UserDataManager.kt
package ai.founderos.mobiletracker.example

import android.content.Context

/**
 * Manages persistence of user data (userId, name, email) across app sessions
 * 
 * Stores user information in SharedPreferences so it can be restored when the app reopens,
 * unless the user explicitly resets the session or all data.
 */
object UserDataManager {
    private const val PREFS_NAME = "user_data_prefs"
    private const val KEY_USER_ID = "user_id"
    private const val KEY_USER_NAME = "user_name"
    private const val KEY_USER_EMAIL = "user_email"
    
    /**
     * Save user data to SharedPreferences
     */
    fun saveUserData(context: Context, userId: String, name: String, email: String) {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        prefs.edit().apply {
            putString(KEY_USER_ID, userId)
            putString(KEY_USER_NAME, name)
            putString(KEY_USER_EMAIL, email)
            apply()
        }
    }
    
    /**
     * Load user data from SharedPreferences
     */
    fun loadUserData(context: Context): UserData? {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        
        val userId = prefs.getString(KEY_USER_ID, null) ?: return null
        val name = prefs.getString(KEY_USER_NAME, "") ?: ""
        val email = prefs.getString(KEY_USER_EMAIL, "") ?: ""
        
        return UserData(userId = userId, name = name, email = email)
    }
    
    /**
     * Clear user data from SharedPreferences
     */
    fun clearUserData(context: Context) {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        prefs.edit().apply {
            remove(KEY_USER_ID)
            remove(KEY_USER_NAME)
            remove(KEY_USER_EMAIL)
            apply()
        }
    }
}

/**
 * Data class for user information
 */
data class UserData(
    val userId: String,
    val name: String,
    val email: String
)
